core/src/block.rs

```diff
@@ -19,894 +19,614 @@ use iroha_data_model::{block::*, events::prelude::*, transaction::prelude::*};
-    #[version_with_scale(n = 1, versioned = "VersionedPendingBlock")]
-    #[derive(Debug, Clone, Decode, Encode)]
-    pub struct PendingBlock {
-        /// Unix time (in milliseconds) of block forming by a peer.
-        pub timestamp: u128,
-        /// array of transactions, which successfully passed validation and consensus step.
-        pub transactions: Vec<VersionedAcceptedTransaction>,
-        /// Event recommendations.
-        pub event_recommendations: Vec<Event>,
-    }
-
-    impl PendingBlock {
-        /// Create a new [`PendingBlock`] from transactions.
-        #[inline]
-        pub fn new(
-            transactions: Vec<VersionedAcceptedTransaction>,
-            event_recommendations: Vec<Event>,
-        ) -> Self {
-            #[allow(clippy::expect_used)]
-            let timestamp = crate::current_time().as_millis();
-            // TODO: Need to check if the `transactions` vector is empty. It shouldn't be allowed.
-            Self {
-                timestamp,
-                transactions,
-                event_recommendations,
-            }
-        }
+declare_versioned_with_scale!(VersionedPendingBlock 1..2, Debug, Clone, iroha_macro::FromVariant);
+
+/// Transaction data is permanently recorded in chunks called
+/// blocks.
+#[version_with_scale(n = 1, versioned = "VersionedPendingBlock")]
+#[derive(Debug, Clone, Decode, Encode)]
+pub struct PendingBlock {
+    /// Block header
+    pub header: BlockHeader,
+    /// Array of rejected transactions.
+    pub rejected_transactions: Vec<VersionedRejectedTransaction>,
+    /// Array of transactions, which successfully passed validation and consensus step.
+    pub transactions: Vec<VersionedValidTransaction>,
+    /// Event recommendations.
+    pub event_recommendations: Vec<Event>,
+    /// Signatures of peers which approved this block
+    pub signatures: SignaturesOf<Self>,
+}
 
-        /// Chain block with the existing blockchain.
-        pub fn chain(
-            self,
-            height: u64,
-            previous_block_hash: Option<HashOf<VersionedCommittedBlock>>,
-            view_change_index: u64,
-            committed_with_topology: Topology,
-        ) -> ChainedBlock {
-            ChainedBlock {
-                transactions: self.transactions,
-                event_recommendations: self.event_recommendations,
-                header: BlockHeader {
-                    timestamp: self.timestamp,
-                    consensus_estimation: DEFAULT_CONSENSUS_ESTIMATION_MS,
-                    height: height + 1,
-                    view_change_index,
-                    previous_block_hash,
-                    transactions_hash: None,
-                    rejected_transactions_hash: None,
-                    committed_with_topology: committed_with_topology.sorted_peers,
-                },
-            }
-        }
+impl PendingBlock {
+    /// Create a new [`PendingBlock`] from transactions.
+    #[inline]
```

Comment:
This function for sure is not a good `inline` candidate, given the register pressure.

Reply:
I agree. This was a copy/paste/refactor mistake.